lexicon/conversion.py

```diff
@@ -280,3 +279,109 @@ def run(self):
+        :param str orth_mapping_file:
+            orthography mapping file, .json .json.gz .txt .gz
+            in case of text file, one can adjust mapping_delimiter
+        :param bool reverse_mapping:
+            reverse/flip the mapping
+        :param str mapping_delimiter:
+            delimiter of source and target orths
+            if mapping is provided with a plain text file
+        """
+
+        self.set_vis_name("Convert Between Regional Orth Spellings")
+
+        self.bliss_lexicon = bliss_lexicon
+        self.orth_mapping_file = orth_mapping_file
+        self.reverse_mapping = reverse_mapping
+        self.mapping_delimiter = mapping_delimiter
+
+        self.out_bliss_lexicon = self.output_path("lexicon.xml.gz")
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+
+        # load mapping from json or plain text file
+        is_json = self.orth_mapping_file.endswith(".json")
+        is_json = is_json or self.orth_mapping_file.endswith(".json.gz")
+        if is_json:
+            with uopen(self.orth_mapping_file, "rt") as f:
+                mapping = json.load(f)
+            if self.reverse_mapping:
+                mapping = {v: k for k, v in mapping.items()}
+        else:
+            mapping = dict()
+            with uopen(self.orth_mapping_file, "rt") as f:
+                for line in f:
+                    line = line.strip()
+                    if not line or line.startswith("#"):
+                        continue
+                    source, target = line.split(self.mapping_delimiter)
+                    if self.reverse_mapping:
+                        source, target = target, source
+                    mapping[source] = target
+
+        # load input lexicon and build orthography to lemma dict
+        lex = lexicon.Lexicon()
+        lex.load(self.bliss_lexicon.get())
+        orth2lemma = {o: lemma for lemma in lex.lemmata for o in lemma.orth}
+
+        # conversion
+        for source, target in mapping.items():
+
+            # if target lemma already existing
+            if target in orth2lemma:
+                final_lemma = orth2lemma[target]
+
+                if source in orth2lemma:
+                    source_lemma = orth2lemma[source]
+
+                    # add pronunciations of source lemma to final lemma
+                    for phon in source_lemma.phon:
+                        if phon not in final_lemma.phon:
+                            final_lemma.phon.append(phon)
+
+                    # remove source orth from source lemma
+                    if source in source_lemma.orth:
+                        source_lemma.orth.remove(source)
+
+                    # remove source lemma from lexicon if no orth remaining
+                    if not source_lemma.orth:
+                        lex.remove_lemma(source_lemma)
+
+                # add source orth as synt (for LM query) to final lemma
+                final_lemma.synt.append([source])
```

Comment:
maybe check if source already had a synt set and then add this instead the orth.

similarly, if target had no synt defined, then setting it here will overwrite the default behavior of (if empty then use orth) i.e. the target lemma will now always use the source orth for LM lookup and never the original target orth.

With the current level of indentation the source orth will be added to synt of target even if source does not exist. Is this intended? I think it should not be.

Reply:
the idea of conversion job is: we have US-English biased system, especially the lexicon and language model are so, we want to only simply convert the lexicon such that the ASR system will output UK-spelling. 

what done here is basically, make US-spelling as UK-spelling's lemma's synt, such that decoder will search US-spelling in the LM, but output UK-spelling.

Reply:
if a lemma already has some synt, what will happen during decoding when we append US-spelling to the synt list?

Reply:
> we want to only simply convert the lexicon such that the ASR system will output UK-spelling.

Then you should not do any of this, but instead add the UK-spelling as `lemma.eval` token. This way internally the US spelling is used consistently for pronunciation, AM, and LM and only the final output is modified to UK spelling.